linera-witty/src/type_traits/implementations/tests.rs

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Unit tests for implementations of the custom traits for existing types.
+
+use crate::{FakeInstance, InstanceWithMemory, Layout, WitLoad, WitStore};
+use frunk::hlist;
+use std::fmt::Debug;
+
+/// Test roundtrip of a heterogeneous list that doesn't need any internal padding.
+#[test]
+fn hlist_without_padding() {
```

Comment:
Maybe assert the SIZE of the `WitType`s?

Reply:
Good idea. Done inside the `test_memory_roundtrip` function.